### PR TITLE
Create csec-tls.adoc

### PR DESCRIPTION
**Pull request by dmuelle**

### CODE REVIEW

File: ref/general/csec-tls.adoc [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+The initial exchanges of the handshake are asymmetrically encrypted by the client and server by using a pair of keys, public and private.
+The client uses the server's public key to encrypt the data it sends, but only the server's private key can decrypt the data.
+The same is true for data that the server sends to the client.
+However, asymmetric encryption requires considerable computing resources.
+Therefore, client and server use these initial exchanges to generate a shared secret key that is used to symmetrically encrypt communications for the remainder of the session.
+Since this shared key is privately generated by each party, rather than being exchanged between them, there is less chance for key to be exposed to eavesdroppers.
+
+A basic TLS handshake comprises the following high-level steps:
+
+. The TLS client sends a `client hello` message, which includes the TLS version, a list of preferred CypherSuites and a random bite string that is used in subsequent computations.
```
Comment (Charlotte-Holt) [inline ref/general/csec-tls.adoc]: Do you need a comma between "CypherSuites" and "and a random"?

File: ref/general/csec-tls.adoc [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+. The client sends a `finished` message, which is encrypted with the shared secret key and indicates that the client side of the handshake is complete.
+. The server sends a `finished` message, which is encrypted with the shared secret key and indicates that the server side of the handshake is complete.
+. For the remainder of the session, client and server can exchange symmetrically encrypted messages by using the shared secret key.
+
+=== The TLS record
+
+Once the TLS handshake establishes authentication and encryption details, the TLS record protocol handles the transmission and verification of encrypted messages during the session.
+The TLS record reads the messages to be transmitted, breaks them into manageable chunks and encrypts them according to the terms agreed on during the TLS handshake.
```
Comment (Charlotte-Holt) [inline ref/general/csec-tls.adoc]: Do you need a comma between "manageable chunks" and "and encrypts them"?

File: ref/general/csec-tls.adoc [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+The TLS record also specifies a means to compute the message digest so that message integrity can be verified by both parties.
+On the receiving side, the record decrypts the messages and verifies their integrity by comparing the message digests.
+
+== TLS and Open Liberty
+
+You can enable support for TLS in Open Liberty by adding the `transportSecurity-1.0` feature to your `server.xml` file.
```
Comment (Charlotte-Holt) [inline ref/general/csec-tls.adoc]: I'm not clear whether we should use the plain text version here. I see that you link out to it with the plain text version below.

File: ref/general/csec-tls.adoc [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+You can specify the details of your TLS settings by adding the  `ssl`, `sslDefault`, and `keyStore` elements to your `server.mxl` file.
```
Comment (Charlotte-Holt) [inline ref/general/csec-tls.adoc]: Update `server.mxl` to `server.xml`

File: ref/general/csec-tls.adoc [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+For more information, see the link:/docs/ref/feature/#transportSecurity.html[Transport Security feature].
+
+In addition to adding the `transportSecurity-1.0` feature, the `keystore` element must be added to your `server.xml` file.
```
Comment (Charlotte-Holt) [inline ref/general/csec-tls.adoc]: Same thing here as above. I _think_ that we talked about using the plain text name except in code examples?

File: ref/general/csec-tls.adoc [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+The following example shows the minimum configuration to enable TLS.
```
Comment (Charlotte-Holt) [inline ref/general/csec-tls.adoc]: I think end this sentence with a colon to lead in to the example

File: ref/general/csec-tls.adoc [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+[source,java]
+----
+<feature>transportSecurity-1.0</feature>
+
+<keyStore id="defaultKeyStore" password="yourPassword" />
+----
+
+Open liberty creates a keystore password during profile creation and puts it in the server.env file that is in the server's home directory.
```
Comment (Charlotte-Holt) [inline ref/general/csec-tls.adoc]: server.env to `server.env`

File: ref/general/csec-tls.adoc [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+If there is no keystore element for the `defaultKeyStore` file, this password is used to create a keystore file.
```
Comment (Charlotte-Holt) [inline ref/general/csec-tls.adoc]: I think keystore should be in backticks (`keystore`). Do you think "keystore"  before "keystore file" should also be in backticks? `defaultKeyStore` file is below.

File: ref/general/csec-tls.adoc [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+This keystore file is then used as the `defaultKeyStore` file.
+Likewise, if a `defaultKeyStore` entry exists without a password in the `sever.xml` file when the server starts, the password from the server.env file is used to open the file.
```
Comment (Charlotte-Holt) [inline ref/general/csec-tls.adoc]: `sever.xml` to `server.xml`
server.env to `server.env`

File: ref/general/csec-tls.adoc [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+If you don't want to use the generated keystore password, remove the `keystore_password` entry from the server.env file.
```
Comment (Charlotte-Holt) [inline ref/general/csec-tls.adoc]: server.env to `server.env`

File: ref/general/csec-tls.adoc [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+If a default keystore was already generated with the password from the `server.env` file, you might need to remove it.
```
Comment (Charlotte-Holt) [inline ref/general/csec-tls.adoc]: Not clear on whether "keystore" should be in backticks in this last instance.

File: ref/general/csec-tls.adoc (hunk 3 above)
Comment (dmuelle) [reply ref/general/csec-tls.adoc]: I'm still a little unsure of how this should play out- should we _always_ use plain text name? Even whne describing how t enable by adding server.xml? I changed it here

File: ref/general/csec-tls.adoc (hunk 11 above)
Comment (dmuelle) [reply ref/general/csec-tls.adoc]: I hoping to make a distinction between a specific `keystore` file and the more generic reference to keystore as a type of file. This part is adapted from KC which doesn't use codeph for generic references. But if the distinction is not evident should probably rethink

File: ref/general/csec-tls.adoc (hunk 3 above)
Comment (Charlotte-Holt) [reply ref/general/csec-tls.adoc]: Yeah, that's a good point and something to consider for our OL stylistic guidelines. I think it's fine that you changed it. Maybe we can ask Karen/Laura during their review for their opinions.

**Pull request merged**